tests/contest/runtimetest/src/tests.rs

```diff
@@ -546,6 +547,61 @@ pub fn test_io_priority_class(spec: &Spec, io_priority_class: IOPriorityClass) {
     }
 }
 
+pub fn validate_process_user(spec: &Spec) {
+    let process = spec.process().as_ref().unwrap();
+
+    let uid = getuid();
+    let gid = getgid();
+    let current_umask = umask(nix::sys::stat::Mode::empty());
+    umask(current_umask);
+
+    if Uid::from(process.user().uid()) != uid {
+        eprintln!(
+            "error due to uid want {}, got {}",
+            process.user().uid(),
+            uid
+        )
+    }
+
+    if Gid::from(process.user().gid()) != gid {
+        eprintln!(
+            "error due to gid want {}, got {}",
+            process.user().gid(),
+            gid
+        )
+    }
+
+    if let Err(e) = validate_additional_gids(process.user().additional_gids().as_ref().unwrap()) {
+        eprintln!("error additional gids {e}");
+    }
+
+    if Mode::from_bits(process.user().umask().unwrap()).unwrap() != current_umask {
+        eprintln!(
+            "error due to gid want {}, got {:?}",
+            process.user().umask().unwrap(),
+            current_umask
+        )
+    }
+}
+
+// validate_additional_gids function is used to validate additional groups of user
+fn validate_additional_gids(gids: &Vec<u32>) -> std::result::Result<(), std::io::Error> {
+    let groups = getgroups().unwrap();
+
+    for group in groups {
+        for gid in gids {
+            if group != Gid::from(*gid) {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("error additional gid want {}, got {}", gid, group),
+                ));
+            }
+        }
+    }
```

Comment:
I think this check is wrong. Currently it seems to work because we only have a single gid,  but if we are setting two gids, this logic will attempt to check every gid is equal to _every_other_ gid , which will fail.

I think a correct way would be to check both vec have equal size, and then iterate over one vec, checking each value is present in the other vec, i.e

```rust
let expected = ...
let actual = getgroups().unwrap()

if expected.len() != actual.len(){
   // error...
}

for gid in expected{
   if !actual.includes(gid){
      // err...
   }
}
```
Or something like this.

Reply:
I fixed like this.

https://github.com/youki-dev/youki/pull/2978/commits/4cce0a26416498909da602032f65354b99b726b5